File: csharp/client/Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Client.Model
{
    class Reservation
    {
        public int hotelId { get; set; }

        public string checkIn { get; set; }

        public string checkOut { get; set; }

        public int roomNo { get; set; }

        public string customerName { get; set; }

        public string customerEmail { get; set; }

        public string customerAddress { get; set; }

        public int orderNo { get; set; }
    }
}

File: node/model/reservation.ts
import { Schema } from 'mongoose'

export interface Reservation {
    hotelId: Number,
    checkIn: String,
    checkOut: String,
    roomNo: Number,
    customerName: String,
    customerEmail: String,
    customerAddress: String,
    orderNo?: Number,
}

export const reservationSchema: Schema = new Schema<Reservation>({
  hotelId: { type: Number, required: true },
  checkIn: { type: String, required: true },
  checkOut: { type: String, required: true },
  roomNo: { type: Number, required: true },
  customerName: { type: String, required: true }, 
  customerEmail: { type: String, required: true },
  customerAddress: { type: String, required: true },
  orderNo: { type: Number, required: false },
})